dbt/include/redshift/macros/utils/get_columns_spec_ddl.sql

```diff
@@ -7,21 +7,22 @@
   {%- for i in user_provided_columns -%}
     {%- set col = user_provided_columns[i] -%}
     {%- set constraints = col['constraints'] -%}
-    {%- set ns = namespace(not_null_line = '') -%}
+    {%- set ns = namespace(not_null_line='', has_check_constraints=False) -%}
 
     {%- for constraint in constraints -%}
-      {%- if constraint == 'primary key' -%}
+      {%- if constraint.type == 'primary_key' -%}
         {%- do primary_keys.append(col['name']) -%}
-      {%- elif constraint == 'not null' %}
+      {%- elif constraint.type == 'not_null' %}
         {%- set ns.not_null_line = " not null" -%}
+      {%- elif constraint.type == 'check' %}
+        {%- set ns.has_check_constraints = True -%}
       {%- endif -%}
     {%- endfor -%}
 
     {%- set not_null_line = " not null" if not_null_col else "" -%}
```

Comment:
out of scope from this refactor but.. is this line even necessary?

Reply:
Absolutely not. I'll get rid of it.